packages/overlay/src/overlay-stack.ts

```diff
@@ -67,34 +73,43 @@ export class OverlayStack {
         );
     }
 
-    public openOverlay(event: CustomEvent<OverlayOpenDetail>): void {
-        if (this.isOverlayActive(event.detail.content)) return;
+    public async openOverlay(details: OverlayOpenDetail): Promise<boolean> {
+        /* istanbul ignore if */
+        if (this.isOverlayActive(details.content)) return false;
 
-        requestAnimationFrame(() => {
-            const interaction = event.detail.interaction;
-            if (interaction === 'click') {
-                this.closeAllHoverOverlays();
-            } else if (
-                interaction === 'hover' &&
-                this.isClickOverlayActiveForTrigger(event.detail.trigger)
-            ) {
-                // Don't show a hover popover if the click popover is already active
-                return;
+        if (details.delayed) {
+            const promise = this.overlayTimer.openTimer(details.content);
+            const cancelled = await promise;
+            if (cancelled) {
+                return promise;
             }
+        }
 
-            const activeOverlay = ActiveOverlay.create(event, this.root);
-            this.overlays.push(activeOverlay);
-
-            this.onChange(this.overlays);
+        return new Promise((resolve) => {
+            requestAnimationFrame(() => {
```

Comment:
As you've been looking at this closely, do you think we still need the various `rAF`s across this file? Previously, I had added more to make the various functionality more normalized, but seeing it again I should have spent some time understanding fully what they were for as it would make the code here much easier to read without them...

Reply:
I tried removing this rAF, but things failed when I did. I would like to revisit this and understand it better.

Reply:
Thanks for testing it!